pandas/core/internals/blocks.py

```diff
@@ -1405,7 +1414,14 @@ def where(self, other, cond, align=True, errors="raise", try_cast=False, axis=0)
 
         # our where function
         def func(cond, values, other):
-            other = self._try_coerce_args(other)
+
+            if not (
+                (self.is_integer or self.is_bool)
```

Comment:
this is looking messy, any better way here?

Reply:
agreed its messy, but nothing cleaner i could find here.  current thought is that a couple of steps from now we'll be done with getting rid of the try/excepts for coercion, at which point we'll have a better shot at cleaning these up